miniupnpc/connecthostport.c

```diff
@@ -208,7 +208,7 @@ SOCKET connecthostport(const char * host, unsigned short port,
 			PRINT_SOCKET_ERROR("setsockopt");
 		}
 #endif /* #ifdef MINIUPNPC_SET_SOCKET_TIMEOUT */
-		n = connect(s, p->ai_addr, p->ai_addrlen);
+		n = connect(s, p->ai_addr, (int)p->ai_addrlen);
```

Comment:
```
     int
     connect(int s, const struct sockaddr *name, socklen_t namelen);
```
So I'm not sure casting to int is a good idea

Reply:
That kind of issues is unfortunately common in socket code: [connect](https://msdn.microsoft.com/en-us/library/windows/desktop/ms737625(v=vs.85).aspx)  function in Microsoft library uses `int`.

A workaround could be taken [here](https://github.com/ARMmbed/mbedtls/blob/8266acacc8d6e1c65fba9a048f56339d0827b2fe/library/net_sockets.c#L92).
If you need, an additional commit could be made.